src/main/java/com/questhelper/helpers/quests/thepathofglouphrie/ThePathOfGlouphrie.java

```diff
@@ -0,0 +1,654 @@
+			var enterStoreroom = new ObjectStep(this, ObjectID.TUNNEL_49620, new WorldPoint(2608, 4451, 0),
+				"Enter the storeroom to the east in the Tree Gnome Village dungeon");
+
+			enterStoreroomPuzzle = new ConditionalStep(this, climbIntoDungeon, "Get to Yewnock's storeroom");
+			enterStoreroomPuzzle.addStep(inTreeGnomeVillageDungeon, enterStoreroom);
+			enterStoreroomPuzzle.addStep(inTreeGnomeVillageMiddle, squeezeThroughRailing);
+		}
+
+
+		/// Storeroom monolith puzzle
+		solveMonolithPuzzle = new MonolithPuzzle(this);
+
+		clickLectern = new ObjectStep(this, ObjectID.LECTERN_49673, YewnocksPuzzle.regionPoint(24, 28), "Click the lectern and learn about the lore.");
+		lecternWidgetActive = new WidgetTextRequirement(854, 5, "Chapter 1. Bad advice");
+		clickChapter1 = new WidgetStep(this, "Click Chapter 1 to learn about the mysterious stranger", 854, 5);
+		clickChapter2 = new WidgetStep(this, "Click Chapter 2 to learn about the great king's death", 854, 9);
+		clickChapter3 = new WidgetStep(this, "Click Chapter 3 to learn about the old foe", 854, 13);
+
+		watchLoreCutscene = new DetailedQuestStep(this, "Watch the cutscene");
+		learnLore = new ConditionalStep(this, clickLectern, "Learn about the lore. All items left on the ground are lost.");
+		learnLore.addStep(new Conditions(lecternWidgetActive, learnedAboutChapter1, learnedAboutChapter2), clickChapter3);
+		learnLore.addStep(new Conditions(lecternWidgetActive, learnedAboutChapter1), clickChapter2);
+		learnLore.addStep(lecternWidgetActive, clickChapter1);
+		learnLore.addSubSteps(watchLoreCutscene);
+
+		solveYewnocksMachinePuzzle = new YewnocksPuzzle(this);
+		solveYewnocksMachinePuzzle.addSubSteps(enterStoreroomPuzzle);
+		solveYewnocksMachinePuzzle.addSubSteps(enterStoreroomPuzzle.getSteps());
+
+		watchCutscene = new DetailedQuestStep(this, "Watch the cutscene");
+
+		/// Inform King Bolren
+		{
+			// Kill the Evil Creature
+			var kill = new NpcStep(this, NpcID.EVIL_CREATURE_12477, new WorldPoint(2542, 3169, 0), "");
+			var exitStoreroom = new ObjectStep(this, ObjectID.TUNNEL_49623, YewnocksPuzzle.regionPoint(37, 17), "Exit the storeroom");
+			exitStoreroom.addTeleport(teleToBolren);
+			var exitDungeon = new ObjectStep(this, ObjectID.LADDER_5251, new WorldPoint(2597, 4435, 0), "Exit the dungeon");
+			var squeezeThroughRailing = enterTreeGnomeVillageMazeFromMiddle.copy();
+			squeezeThroughRailing.setText("");
+			squeezeThroughRailing.addTeleport(teleToBolren);
+			killEvilCreature = new ConditionalStep(this, kill, "Kill the Evil Creature next to King Bolren");
+			killEvilCreature.addStep(inTreeGnomeVillageDungeon, exitDungeon);
+			killEvilCreature.addStep(inStoreroom, exitStoreroom);
+			killEvilCreature.addStep(new Conditions(LogicType.NOR, inTreeGnomeVillageMiddle), squeezeThroughRailing);
+		}
+
+		// Talk to King Bolren
+		informKingBolren = new NpcStep(this, NpcID.KING_BOLREN, new WorldPoint(2542, 3169, 0), "Talk to King Bolren about your next step");
```

Comment:
A note along the lines of 'you can drop all the shapes you obtained now.' could be good.

Reply:
I added it to the bank step https://github.com/Zoinkwiz/quest-helper/pull/1289/files#diff-a9db96efed8ab7bbdc1685921b4dadb144079ae7562aa0b1d2410c7a95ed036fR62-R63